validation.py

```
@@ -0,0 +1,681 @@
+            errors.extend(stream_errors)
+        
+        is_valid = len(errors) == 0
+        sanitized_data = self._sanitize_shared_store(shared) if is_valid else None
+        
+        return ValidationResult(
+            is_valid=is_valid,
+            errors=errors,
+            sanitized_data=sanitized_data
+        )
+    
+    def _validate_task_requirements(self, task_req: Any) -> List[ValidationError]:
```

Ruff (0.12.2)

Dynamically typed expressions (typing.Any) are disallowed in `task_req`

(ANN401)